package.json

```diff
@@ -123,6 +123,49 @@
         "command": "io.orta.jest.coverage.toggle",
         "title": "Jest: Toggle Coverage Overlay"
       }
+    ],
+    "debuggers": [
+      {
+        "type": "node",
+        "label": "Debug Jest tests using vscode-jest"
+      },
+      {
+        "type": "vscode-jest-tests",
+        "label": "Debug Jest tests using vscode-jest",
+        "languages": ["javascript", "javascriptreact", "typescript", "typescriptreact"],
+        "configurationSnippets": [
+          {
+            "label": "Jest: Default jest configuration",
+            "description": "Running tests by directly running jest",
+            "body": {
+              "type": "node",
+              "name": "vscode-jest-tests",
+              "request": "launch",
+              "program": "${workspaceFolder}/node_modules/jest/bin/jest",
+              "args": ["--runInBand"],
+              "cwd": "${workspaceFolder}",
+              "console": "integratedTerminal",
+              "internalConsoleOptions": "neverOpen"
+            }
+          },
+          {
+            "label": "Jest: create-react-app configuration",
+            "description":
+              "Running jest within apps bootstraped by create-react-app (and similar); you may have to edit `runtimeExecutable`",
+            "body": {
+              "type": "node",
+              "name": "vscode-jest-tests",
+              "request": "launch",
+              "runtimeExecutable": "${workspaceFolder}/node_modules/.bin/react-scripts",
+              "args": ["test", "--runInBand"],
+              "cwd": "${workspaceFolder}",
+              "console": "integratedTerminal",
+              "protocol": "inspector",
+              "internalConsoleOptions": "neverOpen"
+            }
+          }
+        ]
+      }
```

Comment:
hmm... I like the snippets, but not sure about splitting DebugConfigurationProvider... I think the original single DebugConfigurationProvider is cleaner. These snippets should just work with your original DebugConfigurationProvider registered under "node" type, right? As we discussed earlier, the `vscode-jest-tests` is really a **fake** type, our DebugConfigurationProvider is really a node type debug provider, IMHO, better to keep that clarity in code.

Reply:
> Having a single DebugConfigurationProvider would be cleaner, for sure. However we need the vscode-jest-tests provider such that the Debugconfigurationprovider appears as a separate entry when initially creating a launch.json config and selecting the environment.

you can accomplish this via registering the same DebugConfigurationProvider under both "node" and "vscode-jest-tests" (see comments in `extension.ts`), without splitting the DebugConfigurationProvider...

Reply:
But then it doesn't only add the config when creating the launch configuration and selecting `Debug Jest tests using vscode-jest`, but also when selecting `Node.js` instead.

Reply:
true, but not sure why is that a problem...

Reply:
I found it somehow unesthetic, but since this is only the case when the extension is activated (and therefore jest is in use), I just merged them.

Reply:
👍